MFVNeutralino/plugins/AnalysisCuts.cc

```diff
@@ -479,379 +591,620 @@ bool MFVAnalysisCuts::filter(edm::Event& event, const edm::EventSetup& setup) {
+        }
+  }
+  return false;
+}
 
-        case mfv::b_HLT_DoubleJetsC100_DoubleBTagCSV_p014_DoublePFJetsC100MaxDeta1p6 :
-            {
-                if(year != 20161 and year != 20162) return false;
-                if(sel_btags_hard < 2) return false;
 
-                for(int j0 = 0; j0 < njets; ++j0){
-                    if(!jet_hlt_match(mevent, j0) || mevent->jet_pt[j0] < 110) continue;
+//splitting up the satisfieslep trigger into muon/ele trigger 
+bool MFVAnalysisCuts::satisfiesLepTrigger(edm::Handle<MFVEvent> mevent, size_t trig, const edm::EventSetup& setup) { 
+  if(!mevent->pass_hlt(trig)) return false;
+
+  int year = int(MFVNEUTRALINO_YEAR);
+  assert(year == 20161 || year == 20162 || year == 2017 || year == 2018); // in case of race conditions where the compiled macro is invalid...
+
+  int nmuons     = mevent->nmuons();
+  int nelectrons = mevent->nelectrons();
+  int njets      = mevent->njets(20);
+    
+  bool passed_kinematics = false;
+
+  switch(trig){
+  case mfv::b_HLT_Ele27_WPTight_Gsf : //for 2016
+    {
+    if (year != 20161 || year !=20162) return false;
+    for(int ie =0; ie < nelectrons; ++ie){
+      if (mevent->electron_pt[ie] < 30) continue; //for 2016
+      if (mevent->electron_ID[ie][3] == 1) {
+        if (abs(mevent->electron_eta[ie]) < 2.4) { 
+          if (mevent->electron_iso[ie] < 0.10) {
+            passed_kinematics = true;
+          } 
+        }
+      }
+    }
+    return passed_kinematics;
+  }
+  case mfv::b_HLT_Ele35_WPTight_Gsf : //for 2017
+    {
+    if (year != 2017) return false;
+    for(int ie =0; ie < nelectrons; ++ie){
+      if (mevent->electron_pt[ie] < 38) continue; //for 2017
+      if (mevent->electron_ID[ie][3] == 1) {
+        if (abs(mevent->electron_eta[ie]) < 2.4) { 
+          if (mevent->electron_iso[ie] < 0.10) {
+            passed_kinematics = true;
+          } 
+        }
+      }
+    }
+    return passed_kinematics;
+  }
+  case mfv::b_HLT_Ele32_WPTight_Gsf : //for 2018
+    {
+      if (year != 2018) return false;
+      for(int ie =0; ie < nelectrons; ++ie){
+        if (mevent->electron_pt[ie] < 35) continue; //for 2018
+	      if (mevent->electron_ID[ie][3] == 1) {
+	        if (abs(mevent->electron_eta[ie]) < 2.4) { 
+	          if (mevent->electron_iso[ie] < 0.10) {
+	            passed_kinematics = true;
+	          } 
+	        }
+	      }
+      }
+      return passed_kinematics;
+    }
+                    
+  case mfv::b_HLT_IsoMu27 : //for 2016,2017
+    {
+    if (year != 2016 || year != 2017) return false;
+    for(int im =0; im < nmuons; ++im) {
+      if (mevent->muon_pt[im] < 30) continue; //for 2016,2017
+      if (mevent->muon_ID[im][1] == 1) {
+        if (abs(mevent->muon_eta[im]) < 2.4) {
+          if (mevent->muon_iso[im] < 0.15) {
+            passed_kinematics = true;
+          }
+        }
+      }
+    }
+    return passed_kinematics;
+  }
+  case mfv::b_HLT_IsoMu24 : //for 2018
+    {
+      if (year != 2018) return false;
```

Comment:
@abbywarden I am guessing you added this, but not positive. On https://twiki.cern.ch/twiki/bin/viewauth/CMS/HLTPathsRunIIList it looks like Mu24 was available (fully unprescaled) in 2016 as well. I would consider changing to use it for 2016 instead of Mu27!